### ISSUE TITLE
Onboarding UX Moonshot

### ISSUE THREAD

**Issue by lpgeiger** [OLSF/libra#379]
- [x] packaging binaries https://github.com/OLSF/libra/issues/378
- [x] ol-cli https://github.com/OLSF/libra/issues/377
- [x] fullnode payment to not wait until epoch changes https://github.com/OLSF/libra/issues/382
- [x] fullnode can transition to validator node gracefully (without switching yaml)  https://github.com/OLSF/libra/issues/383

This is a proposed workflow for getting a new validator on the network.

This UX is a moonshot.

```
> snapd install 0l-node 0l-cli

// starts a node in background
> 0l fullnode

// creates configs on the node
> 0l val-wizard

// give an account.json to someone to send an intialization transaction
[send transaction]

// once transaction goes through, start mining
> 0l miner start

// You will become a validator on the next epoch boundary (every day at approximately 6pm ET)
```
 There are a number of things that need to happen:

1. Packaging of the executables which is straightforward for linux users (proposing `snapd`) https://github.com/OLSF/libra/issues/378
2. 0l-cli which wraps a number of tools (to replace the Makefile), and downloads artifacts needed https://github.com/OLSF/libra/issues/377
3. Background applications, which don't require systemd configuration or screen

**Comment by liangping** [comment]
Instead of using snap, I think this approach will be simpler.
* Build:
1. build project with `cargo deb`
2. upload to github.com/release with CI

* Install:
```
wget https://github.com/jiro4989/nimjson/releases/download/v1.2.5/nimjson_1.2.5_amd64.deb
sudo dpkg -i ./nimjson*.deb
```

What's your opinion?

**Comment by lpgeiger** [comment]
Ok I think that will works temporarily. I think that some OS dependencies will be needed like libgmp for the VDF, and ideally that should be handled by an installer.

Perhaps as an intermediary solution we can have a shell script to make everything a single step:
https://www.rust-lang.org/tools/install

**Comment by lpgeiger** [comment]
closed with #409
